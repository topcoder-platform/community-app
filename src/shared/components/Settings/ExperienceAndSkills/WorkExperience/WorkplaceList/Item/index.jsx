/**
 * render work Item
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import moment from 'moment';
import Tooltip from 'components/Tooltip';
import { isomorphy } from 'topcoder-react-utils';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/profile', false, /svg/);
}

export default function Item(props) {
  const {
    work,
    index,
    onDeleteItem,
    onEditItem,
  } = props;

  const hasSecondLine = () => {
    if (_.isEmpty(work.timePeriodFrom) && _.isEmpty(work.timePeriodTo)
      && _.isEmpty(work.position) && !work.working) {
      return false;
    }

    return true;
  };

  const current = work.working ? '- Current' : '';

  const deleteTip = (
    <div styleName="tctooltiptext">
      <p>Delete</p>
    </div>
  );

  const editTip = (
    <div styleName="tctooltiptext">
      <p>Edit</p>
    </div>
  );

  return (
    <div styleName="container">
      <div styleName="work-info">
        <div styleName="work-icon">
          <ReactSVG path={assets('./ico-work.svg')} />
        </div>
        <div styleName={`work-parameters${hasSecondLine() ? '' : ' single-line'}`}>
          <div styleName={`parameter-first-line${hasSecondLine() ? '' : ' single-line'}`}>
            { `${work.company}${_.isEmpty(work.industry) ? '' : ` | ${work.industry}`}${_.isEmpty(work.cityTown) ? '' : ` | ${work.cityTown}`}` }
          </div>
          <div styleName="parameter-second-line">
            { `${!_.isEmpty(work.timePeriodFrom) ? moment(work.timePeriodFrom).format('YYYY') : ''}${!_.isEmpty(work.timePeriodTo) ? ` - ${moment(work.timePeriodTo).format('YYYY')}` : ` ${current}`} ${!_.isEmpty(work.position) && (!_.isEmpty(work.timePeriodTo) || !_.isEmpty(work.timePeriodFrom)) ? ' | ' : ''}${!_.isEmpty(work.position) ? `${work.position}` : ''}` }
          </div>
          <div styleName="parameter-second-line-mobile">
            <p>
              {`${!_.isEmpty(work.timePeriodFrom) ? moment(work.timePeriodFrom).format('YYYY') : ''}${!_.isEmpty(work.timePeriodTo) ? ` - ${moment(work.timePeriodTo).format('YYYY')}` : ` ${current}`}`}
            </p>
            <p>
              {`${!_.isEmpty(work.position) ? `${work.position}` : ''}`}
            </p>
          </div>
        </div>
      </div>
      <div styleName="operation-container">
        <Tooltip
          id="edit-tip"
          content={editTip}
          trigger={['hover', 'focus']}
        >
          <a
            styleName="edit"
            onKeyPress={() => onEditItem(index)}
            tabIndex={0}
            role="button"
            onClick={() => onEditItem(index)}
          >
            <img src={assets('./ico-edit.svg')} alt="edit-icon" />
          </a>
        </Tooltip>
        <div styleName="delete-wrapper">
          <Tooltip
            id="delete-tip"
            content={deleteTip}
            trigger={['hover', 'focus']}
          >
            <a
              styleName="delete"
              onKeyPress={() => onDeleteItem(index)}
              tabIndex={0}
              role="button"
              onClick={() => onDeleteItem(index)}
            >
              <img src={assets('./ico-trash.svg')} alt="delete-icon" />
            </a>

          </Tooltip>
        </div>
      </div>
    </div>
  );
}

Item.propTypes = {
  work: PT.shape().isRequired,
  index: PT.number.isRequired,
  onDeleteItem: PT.func.isRequired,
  onEditItem: PT.func.isRequired,
};
