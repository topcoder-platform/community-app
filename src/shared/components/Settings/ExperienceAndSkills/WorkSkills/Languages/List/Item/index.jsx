/**
 * render language Item
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import { isomorphy } from 'topcoder-react-utils';
import Tooltip from 'components/Tooltip';

import languageIcon from 'assets/images/profile/language.png';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/profile', false, /svg/);
}

export default function Item(props) {
  const {
    language,
    index,
    onDeleteItem,
    onEditItem,
  } = props;

  const hasOptions = () => {
    if (_.isEmpty(language.spokenLevel) && _.isEmpty(language.writtenLevel)) {
      return false;
    }

    return true;
  };

  const deleteTip = (
    <div styleName="tctooltiptext">
      <p>Delete</p>
    </div>
  );

  const editTip = (
    <div styleName="tctooltiptext">
      <p>Edit</p>
    </div>
  );

  return (
    <div styleName="container">
      <div styleName="language-info">
        <div styleName="language-icon">
          <img src={languageIcon} alt="language" />
        </div>
        <div styleName={`language-parameters${hasOptions() ? '' : ' single-line'}`}>
          <div styleName={`parameter-first-line${hasOptions() ? '' : ' single-line'}`}>
            { language.language }
          </div>
          {
            (!_.isEmpty(language.spokenLevel) || !_.isEmpty(language.writtenLevel)) && (
              <div styleName="parameter-second-line">
                {
                  (!_.isEmpty(language.spokenLevel) && !_.isEmpty(language.writtenLevel)) && (`Spoken: ${language.spokenLevel.toUpperCase()} | Written: ${language.writtenLevel.toUpperCase()}`)
                }
                {
                  (!_.isEmpty(language.spokenLevel) && _.isEmpty(language.writtenLevel)) && (`Spoken: ${language.spokenLevel.toUpperCase()}`)
                }
                {
                  (_.isEmpty(language.spokenLevel) && !_.isEmpty(language.writtenLevel)) && (`Written: ${language.writtenLevel.toUpperCase()}`)
                }
              </div>
            )
          }
          {
            (!_.isEmpty(language.spokenLevel) || !_.isEmpty(language.writtenLevel)) && (
              <div styleName="parameter-second-line-mobile">
                {
                  !_.isEmpty(language.spokenLevel) && (
                    <p>
                      {`Spoken: ${language.spokenLevel.toUpperCase()}`}
                    </p>
                  )
                }
                {
                  !_.isEmpty(language.writtenLevel) && (
                    <p>
                      {`Written: ${language.writtenLevel.toUpperCase()}`}
                    </p>
                  )
                }
              </div>
            )
          }
        </div>
      </div>
      <div styleName="operation-container">
        <Tooltip
          id="edit-tip"
          content={editTip}
          trigger={['hover', 'focus']}
        >
          <a
            styleName="edit"
            onKeyPress={() => onEditItem(index)}
            tabIndex={0}
            role="button"
            onClick={() => onEditItem(index)}
          >
            <img src={assets('./ico-edit.svg')} alt="edit-icon" />
          </a>
        </Tooltip>
        <div styleName="delete-wrapper">
          <Tooltip
            id="delete-tip"
            content={deleteTip}
            trigger={['hover', 'focus']}
          >
            <a
              styleName="delete"
              onKeyPress={() => onDeleteItem(index)}
              tabIndex={0}
              role="button"
              onClick={() => onDeleteItem(index)}
            >
              <img src={assets('./ico-trash.svg')} alt="delete-icon" />
            </a>

          </Tooltip>
        </div>
      </div>
    </div>
  );
}

Item.propTypes = {
  language: PT.shape().isRequired,
  index: PT.number.isRequired,
  onDeleteItem: PT.func.isRequired,
  onEditItem: PT.func.isRequired,
};
