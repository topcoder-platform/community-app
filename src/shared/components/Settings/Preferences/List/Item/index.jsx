/**
 * Render item of preferences
 */
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import cn from 'classnames';
import { isomorphy } from 'topcoder-react-utils';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/preferences', false, /svg/);
}

export default function Item({
  icon,
  title,
  description,
  link,
  status,
  linkTitle,
  email,
  formLink,
  value,
  isForm,
}) {
  return (
    <div styleName={cn('item-container', status === 'subscribed' && 'subscribed')}>
      <div styleName="body">
        <div styleName="icon-wrapper">
          { assets && assets.keys().includes(`./${icon.toLowerCase()}.svg`) ? <ReactSVG path={assets(`./${icon.toLowerCase()}.svg`)} /> : '' }
        </div>
        <div styleName="info">
          <div styleName="info-first-line">
            { title }
          </div>
          <div styleName="info-second-line">
            { description }
          </div>
        </div>
        {
          status && isForm && (
            <form action={formLink} method="post" id="mc-embedded-subscribe-form" name="mc-embedded-subscribe-form" noValidate>
              <input type="email" value={email} readOnly name="EMAIL" id="mce-EMAIL" />
              <input type="checkbox" id="gdpr_11101" name="gdpr[11101]" value="Y" />
              <input type="text" name="b_65bd5a1857b73643aad556093_28bfd3c062" tabIndex="-1" value="" />
              <input type="submit" value={value} name={value} id="mc-embedded-subscribe" className="submit-button" />
            </form>
          )
        }

        { !isForm && <a href={link} target="_blank" rel="noopener noreferrer" styleName="link-btn"><span>{ linkTitle }</span></a>}
      </div>
    </div>
  );
}

Item.defaultProps = {
  link: '',
  email: '',
  status: '',
  value: '',
  formLink: '',
  isForm: false,
};

Item.propTypes = {
  linkTitle: PT.string.isRequired,
  title: PT.string.isRequired,
  description: PT.string.isRequired,
  status: PT.string,
  link: PT.string,
  isForm: PT.bool,
  formLink: PT.string,
  value: PT.string,
  email: PT.string,
  icon: PT.string.isRequired,
};
