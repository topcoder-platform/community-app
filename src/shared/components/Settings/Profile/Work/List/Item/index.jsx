/**
 * render work Item
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import moment from 'moment';
import { isomorphy } from 'topcoder-react-utils';
import { isEmpty as isDateEmpty } from 'utils/settings';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/profile', false, /svg/);
}

export default function Item(props) {
  const {
    work,
    index,
    onDeleteItem,
    onEditItem,
  } = props;

  const {
    jobDescription,
    technologies,
    jobAchievements,
  } = work;

  const hasSecondLine = () => {
    if (isDateEmpty(work.timePeriodFrom) && isDateEmpty(work.timePeriodTo)
      && _.isEmpty(work.position) && !work.working) {
      return false;
    }

    return true;
  };

  const current = work.working ? '- Current' : '';

  return (
    <div styleName="container">
      <div styleName="work-info">
        <div styleName="work-icon">
          <ReactSVG path={assets('./ico-work.svg')} />
        </div>
        <div styleName={`work-parameters${hasSecondLine() ? '' : ' single-line'}`}>
          <div styleName={`parameter-first-line${hasSecondLine() ? '' : ' single-line'}`}>
            { `${work.company}${_.isEmpty(work.industry) ? '' : ` | ${work.industry}`}${_.isEmpty(work.cityTown) ? '' : ` | ${work.cityTown}`}` }
          </div>
          <div styleName="parameter-second-line">
            { `${!isDateEmpty(work.timePeriodFrom) ? moment(work.timePeriodFrom).format('YYYY') : ''}${!isDateEmpty(work.timePeriodTo) ? ` - ${moment(work.timePeriodTo).format('YYYY')}` : ` ${current}`} ${!_.isEmpty(work.position) && (!isDateEmpty(work.timePeriodTo) || !isDateEmpty(work.timePeriodFrom)) ? ' | ' : ''}${!_.isEmpty(work.position) ? `${work.position}` : ''}` }
          </div>
          <div styleName="parameter-second-line-mobile">
            <p>
              {`${!isDateEmpty(work.timePeriodFrom) ? moment(work.timePeriodFrom).format('YYYY') : ''}${!isDateEmpty(work.timePeriodTo) ? ` - ${moment(work.timePeriodTo).format('YYYY')}` : ` ${current}`}`}
            </p>
            <p>
              {`${!_.isEmpty(work.position) ? `${work.position}` : ''}`}
            </p>
          </div>
        </div>
      </div>
      <div styleName="operation-container">
        <a
          styleName="edit"
          onKeyPress={() => onEditItem(index)}
          tabIndex={0}
          role="button"
          onClick={() => onEditItem(index)}
        >
          <img src={assets('./ico-edit.svg')} alt="edit-icon" />
          <p>
            Edit
          </p>
        </a>
        <a
          styleName="delete"
          onKeyPress={() => onDeleteItem(index)}
          tabIndex={0}
          role="button"
          onClick={() => onDeleteItem(index)}
        >
          <img src={assets('./ico-trash.svg')} alt="delete-icon" />
          <p>
            Delete
          </p>
        </a>
      </div>
      { !_.isEmpty(jobDescription) && jobDescription !== '<p><br></p>'
        && (
          <div styleName="job-description">
            <h4 styleName="title">Job Description</h4>
            <div styleName="html-content" dangerouslySetInnerHTML={{ __html: jobDescription /* eslint-disable-line react/no-danger */ }} />
          </div>
        )
      }
      { !_.isEmpty(technologies)
        && (
          <div styleName="technologies">
            <h4 styleName="title">Technologies</h4>
            <div>{technologies.map(item => item.name).join(', ')}</div>
          </div>
        )
      }
      { !_.isEmpty(jobAchievements) && jobAchievements !== '<p><br></p>'
        && (
          <div styleName="achievements">
            <h4 styleName="title">Outputs and Achievements Within the Role</h4>
            <div styleName="html-content" dangerouslySetInnerHTML={{ __html: jobAchievements /* eslint-disable-line react/no-danger */ }} />
          </div>
        )
      }
    </div>
  );
}

Item.propTypes = {
  work: PT.shape().isRequired,
  index: PT.number.isRequired,
  onDeleteItem: PT.func.isRequired,
  onEditItem: PT.func.isRequired,
};
