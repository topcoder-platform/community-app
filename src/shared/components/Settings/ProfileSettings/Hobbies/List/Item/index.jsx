/**
 * render hobby Item
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import Tooltip from 'components/Tooltip';

import { isomorphy } from 'topcoder-react-utils';


import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/profile', false, /svg/);
}

export default function Item(props) {
  const {
    hobby,
    index,
    onDeleteItem,
    onEditItem,
  } = props;

  const hasSecondLine = () => {
    if (_.isEmpty(hobby.description)) {
      return false;
    }

    return true;
  };

  const deleteTip = (
    <div styleName="tctooltiptext">
      <p>Delete</p>
    </div>
  );

  const editTip = (
    <div styleName="tctooltiptext">
      <p>Edit</p>
    </div>
  );

  return (
    <div styleName="container">
      <div styleName="hobby-info">
        <div styleName="hobby-icon">
          <ReactSVG path={assets('./ico-hobby.svg')} />
        </div>
        <div styleName={`hobby-parameters${hasSecondLine() ? '' : ' single-line'}`}>
          <div styleName={`parameter-first-line${hasSecondLine() ? '' : ' single-line'}`}>
            { hobby.hobby }
          </div>
          {
            hasSecondLine() && (
              <div styleName="parameter-second-line">
                { hobby.description }
              </div>
            )
          }
        </div>
      </div>
      <div styleName="operation-container">
        <Tooltip
          id="edit-tip"
          content={editTip}
          trigger={['hover', 'focus']}
        >
          <a
            styleName="edit"
            onKeyPress={() => onEditItem(index)}
            tabIndex={0}
            role="button"
            onClick={() => onEditItem(index)}
          >
            <img src={assets('./ico-edit.svg')} alt="edit-icon" />
          </a>
        </Tooltip>
        <div styleName="delete-wrapper">
          <Tooltip
            id="delete-tip"
            content={deleteTip}
            trigger={['hover', 'focus']}
          >
            <a
              styleName="delete"
              onKeyPress={() => onDeleteItem(index)}
              tabIndex={0}
              role="button"
              onClick={() => onDeleteItem(index)}
            >
              <img src={assets('./ico-trash.svg')} alt="delete-icon" />
            </a>

          </Tooltip>
        </div>
      </div>
    </div>
  );
}

Item.propTypes = {
  hobby: PT.shape().isRequired,
  index: PT.number.isRequired,
  onDeleteItem: PT.func.isRequired,
  onEditItem: PT.func.isRequired,
};
