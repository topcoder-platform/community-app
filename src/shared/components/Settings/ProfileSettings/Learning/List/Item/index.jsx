/**
 * render education Item
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import moment from 'moment';
import { isomorphy } from 'topcoder-react-utils';
import Tooltip from 'components/Tooltip';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/profile', false, /svg/);
}

export default function Item(props) {
  const {
    education,
    index,
    onDeleteItem,
    onEditItem,
  } = props;

  const hasSecondLine = () => {
    if (_.isEmpty(education.timePeriodFrom) && _.isEmpty(education.timePeriodTo)
      && !education.graduated) {
      return false;
    }

    return true;
  };

  const getDate = () => {
    let start = '';
    if (!_.isEmpty(education.timePeriodFrom)) {
      start = moment(education.timePeriodFrom).format('YYYY');
    }
    let end = '';
    if (!_.isEmpty(education.timePeriodTo)) {
      end = moment(education.timePeriodTo).format('YYYY');
    }

    if (_.isEmpty(start) && _.isEmpty(end)) {
      return '';
    }

    if (!_.isEmpty(start) && !_.isEmpty(end)) {
      return `${start} - ${end} `;
    }

    if (!_.isEmpty(start) && _.isEmpty(end)) {
      return `${start} `;
    }

    if (_.isEmpty(start) && !_.isEmpty(end)) {
      return `${end} `;
    }

    return '';
  };

  const getGraduated = () => {
    const date = getDate();
    if (!_.isEmpty(date)) {
      if (education.graduated) {
        return '| Graduated';
      }

      return '';
    }

    if (education.graduated) {
      return 'Graduated';
    }
    return '';
  };

  const deleteTip = (
    <div styleName="tctooltiptext">
      <p>Delete</p>
    </div>
  );

  const editTip = (
    <div styleName="tctooltiptext">
      <p>Edit</p>
    </div>
  );

  return (
    <div styleName="container">
      <div styleName="education-info">
        <div styleName="education-icon">
          <ReactSVG path={assets('./ico-education.svg')} />
        </div>
        <div styleName={`education-parameters${hasSecondLine() ? '' : ' single-line'}`}>
          <div styleName={`parameter-first-line${hasSecondLine() ? '' : ' single-line'}`}>
            { education.major ? `${education.schoolCollegeName} | ${education.major}` : `${education.schoolCollegeName}`}
          </div>
          {
            hasSecondLine() && (
              <React.Fragment>
                <div styleName="parameter-second-line">
                  {
                    `${getDate()}${getGraduated()}`
                  }
                </div>
                <div styleName="parameter-second-line-mobile">
                  {
                    !_.isEmpty(getDate()) && (
                      <p>
                        {`${getDate()}`}
                      </p>
                    )
                  }
                  {
                    education.graduated && (
                      <p>
                        Graduated
                      </p>
                    )
                  }
                </div>
              </React.Fragment>
            )
          }
        </div>
      </div>
      <div styleName="operation-container">
        <Tooltip
          id="edit-tip"
          content={editTip}
          trigger={['hover', 'focus']}
        >
          <a
            styleName="edit"
            onKeyPress={() => onEditItem(index)}
            tabIndex={0}
            role="button"
            onClick={() => onEditItem(index)}
          >
            <img src={assets('./ico-edit.svg')} alt="edit-icon" />
          </a>
        </Tooltip>
        <div styleName="delete-wrapper">
          <Tooltip
            id="delete-tip"
            content={deleteTip}
            trigger={['hover', 'focus']}
          >
            <a
              styleName="delete"
              onKeyPress={() => onDeleteItem(index)}
              tabIndex={0}
              role="button"
              onClick={() => onDeleteItem(index)}
            >
              <img src={assets('./ico-trash.svg')} alt="delete-icon" />
            </a>

          </Tooltip>
        </div>
      </div>
    </div>
  );
}

Item.propTypes = {
  education: PT.shape().isRequired,
  index: PT.number.isRequired,
  onDeleteItem: PT.func.isRequired,
  onEditItem: PT.func.isRequired,
};
