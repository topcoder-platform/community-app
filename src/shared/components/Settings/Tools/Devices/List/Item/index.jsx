/**
 * render device item.
 */
import _ from 'lodash';
import React from 'react';
import PT from 'prop-types';
import ReactSVG from 'react-svg';
import cn from 'classnames';
import { isomorphy } from 'topcoder-react-utils';
import Tooltip from 'components/Tooltip';

import './styles.scss';

let assets;
if (isomorphy.isClientSide()) {
  assets = require.context('assets/images/tools/device-types', false, /svg/);
}

export default function Item(props) {
  const {
    device,
    index,
    isEditing,
    onDeleteItem,
    onEditItem,
  } = props;

  const hasModel = !_.isEmpty(device.model);
  const secondLineItems = [];
  if (device.manufacturer) {
    secondLineItems.push(device.manufacturer);
  }
  if (device.operatingSystem) {
    secondLineItems.push(device.operatingSystem);
  }

  secondLineItems.push(device.deviceType);
  const secondLine = secondLineItems.join(' | ');

  const deleteTip = (
    <div styleName="tctooltiptext">
      <p>Delete</p>
    </div>
  );

  const editTip = (
    <div styleName="tctooltiptext">
      <p>Edit</p>
    </div>
  );

  return (
    <div styleName={cn('container', { isEditing })}>
      <div styleName="device-info">
        <div styleName="device-icon">
          { assets && assets.keys().includes(`./${device.deviceType.toLowerCase()}.svg`) ? <ReactSVG path={assets(`./${device.deviceType.toLowerCase()}.svg`)} /> : '' }
        </div>
        <div styleName="device-parameters">
          {hasModel ? (
            <div>
              {device.model}
            </div>
          ) : null}
          <div styleName="parameter-second-line">
            {secondLine}
          </div>
        </div>
      </div>
      <div styleName="operation-container">
        <Tooltip
          id="edit-tip"
          content={editTip}
          trigger={['hover', 'focus']}
        >
          <a
            styleName="edit"
            onKeyPress={() => onEditItem(index)}
            tabIndex={0}
            role="button"
            onClick={() => onEditItem(index)}
          >
            <img src={assets('./ico-edit.svg')} alt="edit-icon" />
          </a>
        </Tooltip>
        <div styleName="delete-wrapper">
          <Tooltip
            id="delete-tip"
            content={deleteTip}
            trigger={['hover', 'focus']}
          >
            <a
              styleName="delete"
              onKeyPress={() => onDeleteItem(index)}
              tabIndex={0}
              role="button"
              onClick={() => onDeleteItem(index)}
            >
              <img src={assets('./trash.svg')} alt="delete-icon" />
            </a>

          </Tooltip>
        </div>
      </div>
    </div>
  );
}

Item.propTypes = {
  device: PT.shape().isRequired,
  index: PT.number.isRequired,
  isEditing: PT.bool.isRequired,
  onDeleteItem: PT.func.isRequired,
  onEditItem: PT.func.isRequired,
};
