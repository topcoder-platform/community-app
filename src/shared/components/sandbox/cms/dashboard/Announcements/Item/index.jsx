import moment from 'moment-timezone';
import PT from 'prop-types';
import React from 'react';

import './style.scss';

export default function Item({
  height,
  item,
  level,
  top,
}) {
  const {
    endDate,
    startDate,
    title,
  } = item.fields;
  return (
    <a
      href={`/community-app-assets/api/edit-contentful-entry/${item.sys.id}`}
      styleName="container"
      style={{
        left: 250 + (20 * level),
        minHeight: height,
        right: 20 * level,
        top,
      }}
      target="_blank"
    >
      <h1 styleName="title">{title}</h1>
      <div>Start Date: {moment(startDate).toString()}</div>
      <div>End Date: {moment(endDate).toString()}</div>
    </a>
  );
}

Item.propTypes = {
  height: PT.number.isRequired,
  item: PT.shape({
    endDate: PT.string.isRequired,
    startDate: PT.string.isRequired,
    title: PT.string.isRequired,
  }).isRequired,
  level: PT.number.isRequired,
  top: PT.number.isRequired,
};
