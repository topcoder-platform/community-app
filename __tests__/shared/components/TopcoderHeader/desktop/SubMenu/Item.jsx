import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import Item from 'components/TopcoderHeader/desktop/SubMenu/Item';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <Item
      currentSubMenuTitle="title"
      icon={<div />}
      title="title"
      link="menu url"
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <Item
      currentSubMenuTitle="another title"
      icon={<div />}
      title="title"
      link="menu url"
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
