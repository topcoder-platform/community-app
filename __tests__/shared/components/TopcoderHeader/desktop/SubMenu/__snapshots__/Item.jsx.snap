// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<li
  className="src-shared-components-TopcoderHeader-desktop-SubMenu-Item-___style__item___2Q-ud src-shared-components-TopcoderHeader-desktop-SubMenu-Item-___style__current___38PHh"
  role="button"
  tabIndex={0}
>
  <Link
    enforceA={false}
    openNewTab={false}
    to="menu url"
  >
    <div
      aria-label="title"
    />
    title
  </Link>
</li>
`;

exports[`Matches shallow shapshot 2`] = `
<li
  className="src-shared-components-TopcoderHeader-desktop-SubMenu-Item-___style__item___2Q-ud"
  role="button"
  tabIndex={0}
>
  <Link
    enforceA={false}
    openNewTab={false}
    to="menu url"
  >
    <div
      aria-label="title"
    />
    title
  </Link>
</li>
`;
